programs/stake_api/src/stake_instruction.rs

```diff
@@ -164,28 +164,20 @@ pub fn create_stake_account_and_delegate_stake(
     instructions
 }
 
-// for instructions that whose authorized signer may differ from the account's pubkey
-fn metas_for_authorized_signer(
-    account_pubkey: &Pubkey,
-    authorized_signer: &Pubkey, // currently authorized
-    other_params: &[AccountMeta],
+// for instructions whose authorized signer may already be in account parameters
+fn metas_with_signer(
```

Comment:
I agree with an earlier comment. We need a word for "signer-only", where there's no need for a debit account (or an account at all!).  "authorizer"?

I think this function is trying to do too much. How about just an `add_authorizer(&mut Vec<AccountMeta>, &Pubkey) -> ()`?

Reply:
it's not allowed to add an authorizer that's already listed or else TX compilation fails

Reply:
Just saying to remove the moves:

```rust
// Set the meta with the given pubkey as a signer, otherwise add it as a credit-only meta.
fn add_authorizer(
    metas: &mut Vec<AccountMeta>, // parameter metas, in order
    signer: &Pubkey,             // might already appear in paramters
) {
    for meta in &mut metas {
        if &meta.pubkey == signer {
            meta.is_signer = true; // found it, we're done
            return;
        }
    }
    // signer wasn't in metas, append it after normal parameters
    metas.push(AccountMeta::new_credit_only(*signer, true));
}
```

Reply:
we read moves as copies?